extensions/amp-ad/0.1/amp-ad-ui.js

```diff
@@ -71,6 +72,19 @@ export class AmpAdUIHandler {
       return;
     }
 
+    const flyingCarpetElements =
+      ancestorElementsByTag(this.element_, 'amp-fx-flying-carpet');
+    if (flyingCarpetElements.length > 0) {
+      // Special case: force collapse ad if only child of a flying carpet.
+      const flyingCarpetElement = flyingCarpetElements[0];
+      flyingCarpetElement.getImpl().then(implementation => {
+        if (implementation.children_.length === 1) {
```

Comment:
Can't access private variable of implementation. Also we shouldn't collapse in this case
```
<amp-fx-flying-carpet>
  <div>
    <amp-ad></amp-ad>
    <amp-img></amp-img>
  </div>
</amp-fx-flying-carpet>
```

Could you please verify the `children_.length` equals to 1 in this case? If so, I propose we add another restriction here to check if the `<amp-ad>` is the only direct child of `<amp-fx-flying-carpet>`

Reply:
Oops! 😂 Yes I can verify that